src/main/java/step3/domain/LottoGame.java

```diff
@@ -0,0 +1,106 @@
+package step3.domain;
+
+import step3.LottoException;
+import step3.constants.PrizeGrade;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static step3.configuration.LottoConfig.NUMBER_COUNT_PER_GAME;
+import static step3.constants.MessageConstant.NUMBER_COUNT_SHOULD_BE_N;
+
+public class LottoGame {
+
+	private final Set<LottoNumber> lottoNumbers;
+
+	private static final int ZERO = 0;
+	private static final int MATCH = 1;
+	private static final int NON_MATCH = ZERO;
+
+	private static final String OPEN_PARENTHESIS = "[";
+	private static final String CLOSE_PARENTHESIS = "]";
+	private static final String DELIMITER = ", ";
+
+	public LottoGame(List<LottoNumber> lottoNumberPool) {
+		this.lottoNumbers = lottoNumberPool.stream().limit(NUMBER_COUNT_PER_GAME).collect(Collectors.toSet());
+	}
+
+	public LottoGame(String[] numbers) {
+		this.lottoNumbers = createLottoNumbers(numbers);
+	}
+
+	boolean contains(LottoNumber lottoNumber) {
+		return this.lottoNumbers.contains(lottoNumber);
+	}
+
+
+	public PrizeGrade confirmPrize(PrizeInfo prize) {
+		return PrizeGrade.of(getMatchCount(prize), isMatchBonusNumber(prize));
+	}
+
+	private Set<LottoNumber> createLottoNumbers(String[] numbers) {
+
+		Set<LottoNumber> lottoNumbers = Arrays.stream(numbers)
+											.map(LottoNumber::new)
+											.collect(Collectors.toSet());
+
+		if (!isValidNumberCount(lottoNumbers)) {
+			throw new LottoException(String.format(NUMBER_COUNT_SHOULD_BE_N, NUMBER_COUNT_PER_GAME));
+		}
+		return lottoNumbers;
+	}
+
+	private boolean isValidNumberCount(Set<LottoNumber> lottoNumbers) {
+		return lottoNumbers.size() == NUMBER_COUNT_PER_GAME;
+	}
+
+	private int getMatchCount(PrizeInfo prizeInfo) {
+		return this.lottoNumbers
+				.stream()
+				.map(lottoNumber -> {
+					if(prizeInfo.contains(lottoNumber)) {
+						return MATCH;
+					}
+					return NON_MATCH;
+				})
+				.reduce(ZERO, Integer::sum);
+
```

Comment:
다음과 같이 변경해볼 수 있을 것 같네요 :)
```suggestion
		return (int) this.lottoNumbers
				 .stream()
				 .filter(prizeInfo::contains)
			 	 .count();
```